doc/source/raysgd/v2/migration-guide.rst

```diff
@@ -0,0 +1,356 @@
+
+For example, to get the final model:
+
+**SGD v1**
+
+.. code-block:: python
+
+    from ray.util.sgd import TorchTrainer, TrainingOperator
+
+    class MyTrainingOperator(TrainingOperator):
+       ...
+
+    trainer = TorchTrainer(training_operator_cls=MyTrainingOperator, num_workers=2)
+
+    trainer.train()
+
+    trained_model = trainer.get_model()
+
+**SGD v2**
+
+.. code-block:: python
+
+    from ray.util.sgd.v2 import Trainer
+
+    def train_func():
+       model = Net()
+       trainer_loader = MyDataset()
+       for batch in train_loader:
+           model.train(batch)
+
+       return model
+
+    trainer = Trainer(backend="torch")
+    trainer.start()
+    results = trainer.run(train_func, num_workers=2)
+    assert len(results) == 2
+    trained_model = results[0]
+
+Intermediate Reporting
+~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to access any values *during* the training process, you can do so via ``sgd.report()``. You can pass in any values to ``sgd.report()`` and these values from all workers will be sent to any callbacks passed into your ``Trainer``.
+
+**SGD v1**
+
+.. code-block:: python
+
+    from ray.util.sgd import TorchTrainer, TrainingOperator
+
+    class MyTrainingOperator(TrainingOperator):
+       ...
+
+    trainer = TorchTrainer(training_operator_cls=MyTrainingOperator, num_workers=2)
+
+    for _ in range(3):
+        print(trainer.train(reduce_results=False))
+
+    final_model = trainer.get_model()
+
+
+**SGD v2**
+
+.. code-block:: python
+
+   import ray.util.sgd.v2 as sgd
+   from ray.util.sgd.v2 import SGDCallback, Trainer
+   from typing import List, Dict
+
+   class PrintingCallback(SGDCallback):
+       def handle_result(self, results: List[Dict], **info):
+           print(results)
+
+   def train_func():
+       for i in range(3):
+           sgd.report(epoch=i)
+
+   trainer = Trainer(backend="torch", num_workers=2)
+   trainer.start()
+   result = trainer.run(
+       train_func,
+       callbacks=[PrintingCallback()]
+   )
+   # [{'epoch': 0, '_timestamp': 1630471763, '_time_this_iter_s': 0.0020279884338378906, '_training_iteration': 1}, {'epoch': 0, '_timestamp': 1630471763, '_time_this_iter_s': 0.0014922618865966797, '_training_iteration': 1}]
+   # [{'epoch': 1, '_timestamp': 1630471763, '_time_this_iter_s': 0.0008401870727539062, '_training_iteration': 2}, {'epoch': 1, '_timestamp': 1630471763, '_time_this_iter_s': 0.0007486343383789062, '_training_iteration': 2}]
+   # [{'epoch': 2, '_timestamp': 1630471763, '_time_this_iter_s': 0.0014500617980957031, '_training_iteration': 3}, {'epoch': 2, '_timestamp': 1630471763, '_time_this_iter_s': 0.0015292167663574219, '_training_iteration': 3}]
+   trainer.shutdown()
+
+See the :ref:`v2 User Guide <sgd-user-guide>` for more details.
+
+Checkpointing
```

Comment:
Mention the V1 API  that would be migrated.

Reply:
Done